compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -335,8 +335,8 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for CompileTimeInterpreter<'mir,
     }
 
     #[inline(always)]
-    fn enforce_validity(ecx: &InterpCx<'mir, 'tcx, Self>) -> bool {
-        ecx.tcx.sess.opts.unstable_opts.extra_const_ub_checks
+    fn enforce_validity(ecx: &InterpCx<'mir, 'tcx, Self>, layout: TyAndLayout<'tcx>) -> bool {
+        ecx.tcx.sess.opts.unstable_opts.extra_const_ub_checks || layout.abi.is_uninhabited()
```

Comment:
Oh, nice!  That also means future PRs can easily try tuning this to catch more things where validation is cheap enough that perf says it's ok.  (Like maybe checking just `bool`+`char`, or `.abi.is_scalar()`.)

Reply:
Sadly perf being clean mostly means we don't have stress tests going through that path